plugins/plugin-chart-echarts/src/Tree/transformProps.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { ChartProps, getMetricLabel, DataRecord } from '@superset-ui/core';
+import { EChartsOption, TreeSeriesOption } from 'echarts';
+import { TreeSeriesNodeItemOption } from 'echarts/types/src/chart/tree/TreeSeries';
+import { EchartsTreeFormData, DEFAULT_FORM_DATA as DEFAULT_GRAPH_FORM_DATA } from './types';
+import { DEFAULT_TREE_SERIES_OPTION, tooltip } from './constants';
+import { EchartsProps } from '../types';
+import { OptionDataValue, OptionName } from 'echarts/types/src/util/types';
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queriesData } = chartProps;
+  const data: DataRecord[] = queriesData[0].data || [];
+
+  const {
+    id,
+    relation,
+    name,
+    rootNode,
+    metric = '',
+    layout,
+    orient,
+    symbol,
+    symbolSize,
+    roam,
+  }: EchartsTreeFormData = { ...DEFAULT_GRAPH_FORM_DATA, ...formData };
+
+  const metricLabel = getMetricLabel(metric);
+  const tree: TreeSeriesNodeItemOption = { name: rootNode, children: [] };
+  const indexMap: { [name: string]: number } = {};
+  let rootNodeId: null | string = null;
+  for (let i = 0; i < data.length; i++) {
+    const nodeId = data[i][id] as string;
+    indexMap[nodeId] = i;
+    data[i].children = [] as any;
+    if (data[i][name] == rootNode) {
+      rootNodeId = nodeId;
+    }
+  }
+
+  function getNode(
+    name: OptionName,
+    children: TreeSeriesNodeItemOption[],
+    value: OptionDataValue,
+  ): TreeSeriesNodeItemOption {
+    return { name, children, value };
+  }
```

Comment:
As this function isn't really doing that much, maybe we can just do `{ name, children, value }` where this is used instead of calling `getNode(name, children, value)` (I think TS should be able to validate the type equally well this way). Alternatively maybe call this `createNode`, as calling it `getNode` can be interpreted as getting an existing object.